### ISSUE TITLE
JedisCluster can't get a resource

### ISSUE THREAD

**Issue by WonMe** [xetorthio/jedis#1374]
### Expected behavior

I create a JedisCluster instance,it's ok ,but when i try to use it to get a key from redis server,it throw a exception ：
redis.clients.jedis.exceptions.JedisConnectionException: Could not get a resource from the pool
	at redis.clients.util.Pool.getResource(Pool.java:53)
	at redis.clients.jedis.JedisPool.getResource(JedisPool.java:226)
	at redis.clients.jedis.JedisSlotBasedConnectionHandler.getConnectionFromSlot(JedisSlotBasedConnectionHandler.java:66)
	at redis.clients.jedis.JedisClusterCommand.runWithRetries(JedisClusterCommand.java:116)
	at redis.clients.jedis.JedisClusterCommand.runWithRetries(JedisClusterCommand.java:141)
	at redis.clients.jedis.JedisClusterCommand.runWithRetries(JedisClusterCommand.java:141)
	at redis.clients.jedis.JedisClusterCommand.runWithRetries(JedisClusterCommand.java:141)
	at redis.clients.jedis.JedisClusterCommand.runWithRetries(JedisClusterCommand.java:141)
	at redis.clients.jedis.JedisClusterCommand.run(JedisClusterCommand.java:31)
	at redis.clients.jedis.JedisCluster.get(JedisCluster.java:124)
	at xiwan.RedisCluster.get(RedisCluster.java:104)
	at xiwan.RedisCluster.main(RedisCluster.java:120)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: redis.clients.jedis.exceptions.JedisConnectionException: java.net.ConnectException: Connection refused: connect
	at redis.clients.jedis.Connection.connect(Connection.java:207)
	at redis.clients.jedis.BinaryClient.connect(BinaryClient.java:93)
	at redis.clients.jedis.BinaryJedis.connect(BinaryJedis.java:1767)
	at redis.clients.jedis.JedisFactory.makeObject(JedisFactory.java:106)
	at org.apache.commons.pool2.impl.GenericObjectPool.create(GenericObjectPool.java:868)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:435)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:363)
	at redis.clients.util.Pool.getResource(Pool.java:49)
	... 16 more
Caused by: java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.waitForConnect(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:85)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at redis.clients.jedis.Connection.connect(Connection.java:184)
	... 23 more

### Actual behavior

Write here what happens instead ...

### Steps to reproduce:

Please create a reproducible case of your problem. Make sure
that case repeats consistently and it's not random
1.
2.
3.

#### Jedis version:
2.9.0
#### Redis version:
3.2.3
#### Java version:
1.8

**Comment by marcosnils** [comment]
@WonMe as the issue template states, can you please provide a reproducible case (snippet preferably) so we can help out?. According to the stacktrace seems like your java app can't connect to the redis instance. Are you sure you're specifying all the connection parameters correctly?

**Comment by marcosnils** [comment]
@WonMe check https://github.com/antirez/redis/issues/2527

**Comment by alikiller1** [comment]
use this
bind 127.0.0.1 192.168.203130

**Comment by sgarg7** [comment]
@marcosnils I have the exact stack trace with a redis cluster but I'm not using a virtual machine. I'm even able to fetch the cluster nodes using `_jedisClusterClient.getClusterNodes()` but when I try to read or write to the cluster, I get the exception above. Would you happen to know why that might be?

**Comment by amitkumar912** [comment]
Okay I just got to the bottom of this with @sgarg7 .

So what we did was just follow the [Redis Cluster creation tutorial](https://redis.io/topics/cluster-tutorial) to a T. When it tells you to create the cluster using `redis-tribe.rb create` it has you specify 127.0.0.1 for all the nodes. 

When you are connecting to Jedis, it calls `CLUSTER SLOTS`, which returns 127.0.0.1 coming from the node cluster config. And when Jedis is run remotely it tries to connect to your local pc, which probably isn't running Redis. 

So on your Redis Cluster's Configuration (file nodes.conf on every node), you have it set to the hosts IP address instead of `127.0.0.1`.
